homeassistant/components/klyqa/datacoordinator.py

```diff
@@ -0,0 +1,224 @@
+"""Klyqa datacoordinator."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+import socket
+from typing import Any
+
+from klyqa_ctl import klyqa_ctl as api
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_component import (
+    DEFAULT_SCAN_INTERVAL,
+    EntityComponent,
+)
+
+from .const import (
+    CONF_POLLING,
+    CONF_SYNC_ROOMS,
+    EVENT_KLYQA_NEW_LIGHT,
+    EVENT_KLYQA_NEW_LIGHT_GROUP,
+    LOGGER,
+)
+
+
+class HAKlyqaAccount(api.Klyqa_account):  # type: ignore[misc]
+    """HAKlyqaAccount."""
+
+    hass: HomeAssistant | None
+
+    udp: socket.socket
+    tcp: socket.socket
+    polling: bool
+    sync_rooms: bool
+    scan_interval_conf: float
+
+    def __init__(
+        self,
+        udp: Any,
+        tcp: Any,
+        username: str = "",
+        password: str = "",
+        host: str = "",
+        hass: HomeAssistant | None = None,
+        polling: bool = True,
+        sync_rooms: bool = True,
+        scan_interval: float = -1.0,
+    ) -> None:
+        """HAKlyqaAccount."""
+        super().__init__(username, password, host)
+        self.hass = hass
+        self.udp = udp
+        self.tcp = tcp
+        self.polling = polling
+        self.sync_rooms = sync_rooms
+        self.scan_interval_conf = scan_interval
+
+    async def send_to_bulbs(
+        self,
+        args_parsed: Any,
+        args_in: Any,
+        timeout_ms: Any = 5000,
+        async_answer_callback: Any = None,
+    ) -> Any:
+        """Send_to_bulbs."""
+        ret = await super()._send_to_bulbs(
+            args_parsed,
+            args_in,
+            self.udp,
+            self.tcp,
+            timeout_ms=timeout_ms,
+            async_answer_callback=async_answer_callback,
+        )
+        return ret
+
+    # pylint: disable=arguments-differ
+    async def login(self, **kwargs: Any) -> bool:
+        """Login."""
+        ret: bool = await super().login(**kwargs)
+        if ret:
+            await api.async_json_cache(
+                {
+                    CONF_USERNAME: self.username,
+                    CONF_PASSWORD: self.password,
+                    CONF_SCAN_INTERVAL: self.scan_interval_conf,
+                    CONF_SYNC_ROOMS: self.sync_rooms,
+                    CONF_POLLING: self.polling,
+                    CONF_HOST: self.host,
+                },
+                "last.klyqa_integration_data.cache.json",
+            )
+        return ret
+
+    async def update_account(self) -> bool:
+        """Update_account."""
+
+        await self.request_account_settings()
+        if (
+            self.hass
+            and EVENT_KLYQA_NEW_LIGHT
+            in self.hass.bus._listeners  # pylint: disable=protected-access
```

Comment:
You shouldn't be accessing this, it is a protected variable (please don't suppress the linters, that is not solving the issue).